### PR TITLE
Adds VectorValueExt

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

## What have you changed? (mandatory)

VectorValueExt provides a generic way to push values into a VectorValue.

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

No.

### CODE REVIEW

File: src/coprocessor/codec/data_type/vector.rs [hunk 1]
```diff
@@ -662,16 +675,22 @@ impl_as_slice! { DateTime, as_date_time_slice }
 impl_as_slice! { Duration, as_duration_slice }
 impl_as_slice! { Json, as_json_slice }
 
-macro_rules! impl_push {
-    ($ty:tt, $name:ident) => {
+pub trait VectorValueExt<T: Evaluable> {
```
Comment (AndreMouche) [inline src/coprocessor/codec/data_type/vector.rs]: Could you please add some comments to describe it?

Comment (breeswish) [reply src/coprocessor/codec/data_type/vector.rs]: Done, PTAL again, thanks!

File: src/coprocessor/codec/data_type/vector.rs [hunk 2]
```diff
@@ -662,16 +675,25 @@ impl_as_slice! { DateTime, as_date_time_slice }
 impl_as_slice! { Duration, as_duration_slice }
 impl_as_slice! { Json, as_json_slice }
 
-macro_rules! impl_push {
-    ($ty:tt, $name:ident) => {
+/// Additional `VectorValue` methods available via generics. These methods supports different
```
Comment (rleungx) [inline src/coprocessor/codec/data_type/vector.rs]:
```suggestion
/// Additional `VectorValue` methods available via generics. These methods support different
```

**Comment by breeswish** [comment]
No need any more because https://github.com/tikv/tikv/pull/4533 is merged.

**Pull request closed without merging**